src/controllers/user.controller.ts

```diff
@@ -87,4 +160,56 @@ export class UserController {
   ): Promise<Product[]> {
     return this.recommender.getProductRecommendations(userId);
   }
+
+  @post('/users/login', {
+    responses: {
+      '200': {
+        description: 'Token',
+        content: {
+          'text/plain': {
+            schema: {
+              type: 'string',
+            },
+          },
+        },
+      },
+    },
+  })
+  async login(
+    @requestBody() credentials: Credentials,
+  ): Promise<String | undefined> {
+    // Validate Email
+    if (!isemail.validate(credentials.email)) {
+      throw new HttpErrors.UnprocessableEntity('invalid email');
+    }
+
+    // Validate Password Length
+    if (credentials.password.length < 8) {
+      throw new HttpErrors.UnprocessableEntity(
+        'password must be minimum 8 characters',
+      );
+    }
+
+    // Check if user exists
+    try {
+      const foundUser = await this.userRepository.findOne({
+        where: {email: credentials.email, password: credentials.password},
+      });
+      let token = undefined;
+      if (foundUser) {
+        const currentUser = _.pick(foundUser.toJSON(), ['id', 'email']);
+        // Generate user token using JWT
+        token = await signAsync(currentUser, 'secretforjwt', {
+          expiresIn: 300,
+        });
+      } else {
+        return Promise.reject(
```

Comment:
Just `throw`.

Reply:
We're returning because we want a `401 Unauthorized Error` to be returned when someone logs in with the wrong credentials instead of receiving a generic `Error`.

Reply:
As I see it, the problem is in the way how this code is structured, in particular with this try/catch block:

```ts
try {
  // do everything
  // maybe return Promise.reject()
} catch (err) {
  return Promise.reject(new Error('Wrong Credentials!'));
}
```

I am sure you can find a way how to restructure this block so that we can use `throw` instead of `Promise.reject` and `Promise.resolve`.